src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -53,36 +57,56 @@ public AuthorisedCaseSearchOperation(
     }
 
     @Override
-    public CaseSearchResult execute(CaseSearchRequest caseSearchRequest) {
-        return authorisedCaseDefinitionDataService
-            .getAuthorisedCaseType(caseSearchRequest.getCaseTypeId(), CAN_READ)
-            .map(caseType -> searchCasesAndFilterFieldsByAccess(caseType, caseSearchRequest))
-            .orElse(CaseSearchResult.EMPTY);
+    public CaseSearchResult execute(CrossCaseTypeSearchRequest searchRequest) {
+        List<CaseType> authorisedCaseTypes = getAuthorisedCaseTypes(searchRequest);
```

Comment:
can we pass to getAuthorisedCaseTypes just the list of case type ids rather than the whole searchRequest?